src/resolution/composite-arns-resolver.ts

```diff
@@ -48,63 +50,74 @@ export class CompositeArNSResolver implements NameResolver {
     this.resolvers = resolvers;
     this.cache = cache;
     this.overrides = overrides;
+    this.arnsNamesCache = new ArNSNamesCache({ log });
```

Comment:
I decided to initialize `ArNSNamesCache` here as we will only want to have it working if `CompositeArNSResolver` exists.